lib/compiler/src/beam_doc.erl

```diff
@@ -0,0 +1,170 @@
+-module(beam_doc).
+
+-feature(maybe_expr, enable).
+
+-export([main/1]).
+
+-include_lib("kernel/include/eep48.hrl").
+
+-spec main(term()) -> term().
+main(BeamFile) ->
+    try
+        {ok, _Module, Chunks} = beam_lib:all_chunks(BeamFile),
+        {debug_info_v1, _, {AST, Meta}} = binary_to_term(proplists:get_value("Dbgi", Chunks)),
+        {_, File} = lists:foldl(
+                            fun({attribute, [{generated,true}|_], file, {File, _}}, {false, _}) ->
+                                    {true, File};
+                               (_, File) when is_tuple(File) ->
+                                    File;
+                               ({attribute, _, file, {File,_}}, _) ->
+                                    File;
+                               ({attribute, _Anno, module, _}, File) ->
+                                    {false, File}
+                            end, undefined, AST),
+        Filename = filename:join(proplists:get_value(cwd, Meta, ""), File),
```

Comment:
The compiler should already have this name available somewhere, just need to figure out how to get it.